js/plugins/vertexai/src/gemini.ts

```diff
@@ -517,11 +521,18 @@ export function geminiModel(
         }
       }
 
+      const tools = request.tools?.length
+        ? [{ functionDeclarations: request.tools?.map(toGeminiTool) }]
+        : [];
+
+      // Cannot use tools and function calling at the same time
+      const jsonMode =
+        (request.output?.format === 'json' || !!request.output?.schema) &&
```

Comment:
this implementation does not support passing output schema as model responseSchema?
https://ai.google.dev/gemini-api/docs/structured-output?lang=node#supply-schema-in-config

Reply:
Yeah we need to add this feature in too, do you want these changes on this PR? This PR was in theory just to sync with what went into `main`

Reply:
i'll add as commits here actually, that makes sense